src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -0,0 +1,102 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.BulkLoadRequestModel;
+import bio.terra.service.filedata.exception.BulkLoadControlFileException;
+import bio.terra.service.filedata.google.gcs.GcsBufferedReader;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.load.LoadService;
+import bio.terra.service.load.flight.LoadMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+// Populate the files to be loaded from the incoming array
+public class IngestPopulateFileStateFromFileStep implements Step {
+    // TODO: MAYBE make these configuration items?
+    private static final int BATCH_SIZE = 50;
+    private static final int MAX_BAD_LINES = 5;
+
+    private final LoadService loadService;
+    private final ObjectMapper objectMapper;
+
+    public IngestPopulateFileStateFromFileStep(LoadService loadService, ObjectMapper objectMapper) {
+        this.loadService = loadService;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        FlightMap inputParameters = context.getInputParameters();
+        BulkLoadRequestModel loadRequest =
+            inputParameters.get(JobMapKeys.REQUEST.getKeyName(), BulkLoadRequestModel.class);
+
+        FlightMap workingMap = context.getWorkingMap();
+        UUID loadId = UUID.fromString(workingMap.get(LoadMapKeys.LOAD_ID, String.class));
+
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        List<String> errorDetails = new ArrayList<>();
+
+        try (BufferedReader reader = new GcsBufferedReader(storage, loadRequest.getLoadControlFile())) {
+            long lineCount = 0;
+            List<BulkLoadFileModel> fileList = new ArrayList<>();
+
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                lineCount++;
+
+                try {
+                    BulkLoadFileModel loadFile = objectMapper.readValue(line, BulkLoadFileModel.class);
```

Comment:
General question: Will this complain if there are extra keys in the JSON object being loaded in? For ENCODE, we'll be generating triples of `(s3_path, target_staging_gs_path, target_jade_virtual_path)` for every file we want to ingest. I'm wondering if we could get away with storing all three fields in the same objects, or if we'll need to output two separate streams to GCS.

Reply:
I can set it up so that the mapper would ignore unknown keys. I think I'll make a separate ticket for that and write some tests to make sure it works.